Ask-VA/ 1344 Submit form payload with attachments

ediiotero opened the pull request:
## Are you removing, renaming or moving a folder in this PR?
- [X] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [X] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- Adding a submit data transformer and building formData payload
- _(If bug, how to reproduce)_
- _(What is the solution, why is this the solution)_
- Ask VA team owns the form
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_

## Related issue(s)

- Ticket [1344](https://app.zenhub.com/workspaces/ask-va-647a476551689d06655cc815/issues/gh/department-of-veterans-affairs/ask-va/1344)

## Testing done

- Manual testing done
- Fill out form and submit. Changes pending on the BE

## Screenshots

No UI changes

## What areas of the site does it impact?

Only impacts the Ask VA form

## Acceptance criteria

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [X] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [X] Did you login to a local build and verify all authenticated routes work as expected with a test user

ediiotero committed:
adding submit transformer and post form data

joehall-tw reviewed (dismissed):
LGTM. May want Khoa to review the payload fields separately, but I don't see any issues.

Khoa-V-Nguyen committed:
Update submit-transformer

- Updated payload to a new structure
- Created determineInquiryDetails function
  - Added logic to categorize inquiry based on data properties, including who the question is about and their relationship to the veteran
  - Structured the output to include inquiryAbout, dependentRelationship, veteranRelationship, and levelOfAuthentication fields
  - Incorporated handling for various scenarios such as education benefits, self-inquiry, and inquiries on behalf of someone else
  - Determined level of authentication based on user role and context
  - Implemented default handling for unknown inquiry types

Khoa-V-Nguyen committed:
WIP

ediiotero committed:
update actions and reducer with subtopic id and add askVAStore to submit transformer
